lib/rack/jwt/auth.rb

```diff
@@ -20,34 +23,35 @@ class Auth
         'ES256',
         'ES384',
         'ES512',
-        ('ED25519' if defined?(RbNaCl)),
+        ('ED25519' if defined?(RbNaCl))
       ].compact.freeze
 
-      DEFAULT_ALGORITHM = 'HS256'.freeze
+      DEFAULT_ALGORITHM = 'HS256'
 
       # The last segment gets dropped for 'none' algorithm since there is no
       # signature so both of these patterns are valid. All character chunks
       # are base64url format and periods.
       #   Bearer abc123.abc123.abc123
       #   Bearer abc123.abc123.
       BEARER_TOKEN_REGEX = %r{
-        ^Bearer\s{1}(       # starts with Bearer and a single space
-        [a-zA-Z0-9\-\_]+\.  # 1 or more chars followed by a single period
-        [a-zA-Z0-9\-\_]+\.  # 1 or more chars followed by a single period
-        [a-zA-Z0-9\-\_]*    # 0 or more chars, no trailing chars
+        ^Bearer\s(       # starts with Bearer and a single space
+        [a-zA-Z0-9\-_]+\.  # 1 or more chars followed by a single period
+        [a-zA-Z0-9\-_]+\.  # 1 or more chars followed by a single period
+        [a-zA-Z0-9\-_]*    # 0 or more chars, no trailing chars
         )$
-      }x
+      }x.freeze
 
       # Initialization should fail fast with an ArgumentError
       # if any args are invalid.
       def initialize(app, opts = {})
-        @app     = app
-        @secret  = opts.fetch(:secret, nil)
-        @verify  = opts.fetch(:verify, true)
-        @options = opts.fetch(:options, {})
-        @exclude = opts.fetch(:exclude, [])
-
-        @secret  = @secret.strip if @secret.is_a?(String)
+        @app          = app
+        @secret       = opts.fetch(:secret, nil)
+        @verify       = opts.fetch(:verify, true)
+        @options      = opts.fetch(:options, {})
+        @cookie_name  = @options.fetch(:cookie_name, nil)
```

Comment:
```suggestion
        @cookie_name  = opts.fetch(:cookie_name, nil)
```

I know it's the same thing but makes it easier to read, I think

Reply:
It's different as `@options` is contained within `opts`, but it's a bit awkard to follow, so I'll alter it to use `dig`